### ISSUE TITLE
`JSON.stringify(Error)` not working well

### ISSUE THREAD

**Issue by gurubaksh** [neuledge/computed-types#37]
const UserSchema = Schema({
 username: string.trim().error('expect input to be string'),
 password: string.trim().min(1).error('password should not be empty')
});

type User1 = Type<typeof UserSchema>;
const validator = UserSchema.destruct();

const [err, user]  = await  validator(loginData);
var errString = JSON.stringify(err);
console.log(errString)

not getting error msg. Can please tell me how can i fetch error with fieldname

**Comment by moshest** [comment]
Error are not transform to JSON well.

`Schema` error structure is as the following:
```ts
interface SchemaError extends Error {
  errors: { path: string; error: Error }[];
}
```

In your example you try to `JSON.stringify(new Error('foo'));` which result an empty object instead of the actual error message. 

Check out this for possible solution: https://stackoverflow.com/a/18391400/518153

**Comment by gurubaksh** [comment]
if use below code then does not work -- 

interface SchemaError extends Error {
  errors: { path: string; error: Error }[];
}
const [err, user]  = await  validator(loginData);
	if(err) {
		var bb: SchemaError = err;
		console.log(bb)
}

error msg -- error: TS2741 [ERROR]: Property 'errors' is missing in type 'Error' but required in type 'SchemaError'.
        var bb: SchemaError = err;
            ~~
    at file://

    'errors' is declared here.
      errors: { path: string; error: Error }[];




and if i use below code it does not work -

interface SchemaError {
   errors: { path: object; error: object }[];
}

const [err, user]  = await  validator(loginData);
	if(err) {
		var bb: SchemaError = err;
		console.log(bb)
}

error msg -  

error: TS2741 [ERROR]: Property 'errors' is missing in type 'Error' but required in type 'SchemaError'.
        var bb: SchemaError = err;

 'errors' is declared here.
       errors: { path: object; error: object }[];

 https://stackoverflow.com/a/18391400/518153 works but i believe if i am using third party library then  i should not use custom. otherwise i can write my own validator. 

please let me know is this possible to get error message without writing any custom code because it is a basic requirement 
or it will come in few days or should go for other one if it is not coming in near days.

**Comment by moshest** [comment]
Fixed in #38.

Notice `.destruct();` now return `ValidatorError` type so no need for type assertions.
But if still needed, it's available under `import { ValidatorError } from 'computed-types'`.

Thanks for your feedback!
